[13.0][IMP] website_sale_checkout_skip_payment: Changes on Confirm Order buttons

CarlosRoca13 opened the pull request:
With this changes the 'Process Checkout' button change to 'Confirm' when the partner has selected the option skip_website_checkout_payment. Furthermore, the button text 'Confirm Order' has been changed to 'Confirm'.

cc @Tecnativa TT29736

please @pedrobaeza @victoralmau review this

CarlosRoca13 committed:
[IMP] website_sale_checkout_skip_payment: Changes on Confirm Order Buttons

With this changes the 'Process Checkout' button change to 'Confirm' when the partner has selected the option skip_website_checkout_payment. Furthermore, the button text 'Confirm Order' has been changed to 'Confirm'.

TT29736

pedrobaeza commented:
There should be something different, as the customer has the text "Pay Now". Can you check if there's any difference or other module adding that?

CarlosRoca13 commented:
"Pay Now" appears when the partner asociated to the user has the field skip_website_checkout_payment set as True.

pedrobaeza commented:
It should be the contrary then

CarlosRoca13 commented:
Why?

pedrobaeza commented:
If the payment is skipped (so `skip_website_checkout_payment` = True), then "Pay Now" text has no sense. For this case, "Confirm" is the needed one if I don't understand bad what you are telling to me.

CarlosRoca13 commented:
Ups I thought that I said all the opposite, it's working as you said. When the partner has `skip_website_checkout_payment = False` the button pay now will be shown.

I will work for correct the tests

CarlosRoca13 committed:
[FIX] website_sale_checkout_skip_payment: Error when client is public user

When the client is a public user we got an error, because the request.session.uid is None and the value of the field checkout_skip_payment is not computed.

As we want to compute this field always,the condition is removed to avoid the possible error.

TT29736

CarlosRoca13 commented:
I add the fix on other commit.

chienandalu reviewed (approved):
Tested on runbot :+1:

pedrobaeza commented:
Wow, the number of changes in the PO files is huge. Maybe it's better to leave "Confirm Order" although less generic?

CarlosRoca13 commented:
Okey!

CarlosRoca13 commented:
BTW the changes will be there if we update the po files with the pot, so IDK if it will reduce the changes in po files

pedrobaeza commented:
OK, let's go with this then:

/ocabot merge minor
